crates/primitives/src/header.rs

```diff
@@ -308,21 +325,33 @@ impl<'a> arbitrary::Arbitrary<'a> for SealedHeader {
     }
 }
 
-impl From<Block<EthersH256>> for SealedHeader {
-    fn from(block: Block<EthersH256>) -> Self {
-        let header = Header {
+impl From<&Block<EthersH256>> for Header {
+    fn from(block: &Block<EthersH256>) -> Self {
+        Header {
+            parent_hash: block.parent_hash.0.into(),
             number: block.number.unwrap().as_u64(),
             gas_limit: block.gas_limit.as_u64(),
             difficulty: block.difficulty.into(),
             nonce: block.nonce.unwrap().to_low_u64_be(),
-            extra_data: block.extra_data.0.into(),
+            extra_data: block.extra_data.0.clone().into(),
             state_root: block.state_root.0.into(),
+            transactions_root: block.transactions_root.0.into(),
+            receipts_root: block.receipts_root.0.into(),
             timestamp: block.timestamp.as_u64(),
             mix_hash: block.mix_hash.unwrap().0.into(),
             beneficiary: block.author.unwrap().0.into(),
             base_fee_per_gas: block.base_fee_per_gas.map(|fee| fee.as_u64()),
-            ..Default::default()
-        };
+            ommers_hash: block.uncles_hash.0.into(),
+            gas_used: block.gas_used.as_u64(),
+            withdrawals_root: None,
```

Comment:
@rkrasiuk does this mean we need to update ethers-rs here to have a withdrawals root?